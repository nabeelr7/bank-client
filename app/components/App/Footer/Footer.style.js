import styled from 'styled-components';

export const StyledFooter = styled.footer`
  color: red;
  text-align: center;
  position: absolute;
  width: 100%;
  left: 50%;
  transform: translate(-50%, 0);
  bottom: 10px;
`;
