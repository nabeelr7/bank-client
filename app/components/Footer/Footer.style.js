import styled from 'styled-components';
import { Button } from 'antd';

export const StyledFooter = styled.footer`
  margin: 15px auto;
  max-width: 560px;
  font-size: 13px;
  padding: 0 20px;
`;

export const StyledWarning = styled.div`
  display: flex;
  align-items: center;
  font-weight: 500;
  margin: 20px 0;
  font-size: 14px;
`;

export const StyledList = styled.div``;

export const StyledTip = styled.div`
  color: rgb(229, 0, 0);
  margin: 15px 0;
`;

export const StyledButton = styled(Button)`
  font-weight: 500;
  padding: 0;
  color: inherit;
  height: auto;
  font-size: inherit;
`;
