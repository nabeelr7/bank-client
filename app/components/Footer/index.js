/**
 *
 * Footer
 *
 */

import React from 'react';
import { useDispatch } from 'react-redux';
import { push } from 'connected-react-router';
import {
  StyledFooter,
  StyledButton,
  StyledInfoCircleOutlined,
  StyledWarning,
  StyledTip,
} from './Footer.style';

// import { FormattedMessage } from 'react-intl';
// import messages from './messages';

export default function Footer() {
  const dispatch = useDispatch();

  return (
    <StyledFooter>
      <StyledWarning>
        <StyledInfoCircleOutlined />
        Remember the basic safety rules.
      </StyledWarning>

      <div>
        Before you enter your ID number and access password on your website,
        make sure
        <ul>
          <li>
            Your password is secure. It contains at least 8 characters and
            consists of uppercase and lowercase letters
          </li>
          <li>
            in the address bar or status bar, a padlock is visible at the bottom
            of the browser screen
          </li>
        </ul>
      </div>

      <StyledTip>
        Remember: The bank does not require confirmation of any data, correct
        login or reading of the Banks messages by means of SMS, TAN and / or
        e-mail, or installation of any applications on the users phones or
        computers.
      </StyledTip>

      <div>
        This site uses cookies to provide services at the highest level. By
        clicking or navigating the site, you agree to allow our collection of
        information on through cookies. For more information on security, please
        visit:{' '}
        <StyledButton type="link" onClick={() => dispatch(push('/privacy'))}>
          Privacy rules
        </StyledButton>
      </div>
    </StyledFooter>
  );
}
