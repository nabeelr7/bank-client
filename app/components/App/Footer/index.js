/**
 *
 * Footer
 *
 */

import React from 'react';
import { createStructuredSelector } from 'reselect';
import { makeSelectIsCollapsedSidebar } from 'containers/App/selectors';
import { useSelector } from 'react-redux';
import { StyledFooter } from './Footer.style';

const stateSelector = createStructuredSelector({
  isCollapsedSidebar: makeSelectIsCollapsedSidebar(),
});

export default function Footer() {
  const { isCollapsedSidebar } = useSelector(stateSelector);

  return (
    <StyledFooter collapsed={isCollapsedSidebar}>
      <strong>Bank Application 2.0</strong> | 10.07.20
    </StyledFooter>
  );
}
