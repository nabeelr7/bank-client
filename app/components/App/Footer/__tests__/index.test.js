import React from 'react';
import { render } from '@testing-library/react';
import Footer from '../index';

describe('<Footer />', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Footer />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
