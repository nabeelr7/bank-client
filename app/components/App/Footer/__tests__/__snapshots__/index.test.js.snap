// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match the snapshot 1`] = `
<footer
  class="Footerstyle__StyledFooter-a5b9l3-0 gcmDBN"
>
  <strong>
    Bank Application v2.0
  </strong>
   | 22.05.20
</footer>
`;
